src/js/components/dateselect/index.js

```diff
@@ -0,0 +1,165 @@
+import { isNull, map, range, times } from 'underscore';
+import Backbone from 'backbone';
+import hbs from 'handlebars-inline-precompile';
+import dayjs from 'dayjs';
+
+import intl from 'js/i18n';
+
+import Component from 'js/base/component';
+import Droplist from 'js/components/droplist';
+
+import { LayoutView } from './dateselect_views';
+
+import './date-select.scss';
+
+const i18n = intl.components.dateSelect;
+
+const monthsCollection = new Backbone.Collection(times(12, idx => {
+  const month = dayjs().month(idx);
+
+  return {
+    value: idx,
+    text: `${ month.format('MMMM') } (${ (idx + 1) })`,
+  };
+}));
+
+const selectListOptions = ['buttonText', 'buttonClassName'];
+const SelectList = Droplist.extend({
```

Comment:
I think we want the `isSelectlist: true` so you can type out the date

Reply:
I think this is holding up a release, but otherwise the other changes could be made in a chore.